lib/ansible/plugins/action/include_vars.py

```diff
@@ -40,12 +40,14 @@ def _set_dir_defaults(self):
 
         if isinstance(self.ignore_files, string_types):
             self.ignore_files = self.ignore_files.split()
+            self._display.deprecated(
+                msg="Specifying 'ignore_files' as a string is deprecated.",
+                version="2.23",
+                help_text="Use a list of strings instead.",
+            )
 
-        elif isinstance(self.ignore_files, dict):
-            return {
-                'failed': True,
-                'message': '{0} must be a list'.format(self.ignore_files)
-            }
+        if not isinstance(self.ignore_files, list):
+            raise AnsibleError("Invalid type for 'ignore_files' option, must be a list")
```

Comment:
I'd reword this error message.

Also, provide an `obj` reference so context can be provided to the user in the error message.

```suggestion
            raise AnsibleError("The 'ignore_files' option must be a list.", obj=self.ignore_files)
```

Similar changes can be made to the other error handlers below that are being updated.